runbotics/runbotics-scheduler/src/auth/microsoft.session.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable, Scope } from '@nestjs/common';
+import { Logger } from '../utils/logger';
+import Axios from 'axios';
+export interface AuthData {
+    token: string;
+    refreshToken: string;
+    expires: number;
+}
+
+interface APIToken {
+    access_token: string;
+    refresh_token: string;
+    expires_on: number;
+}
+
+@Injectable({ scope: Scope.DEFAULT })
+export class MicrosoftSession {
+    constructor() { }
+    private readonly logger = new Logger(MicrosoftSession.name);
+    private session: AuthData = {
```

Comment:
`AuthData` type does not accept undefined fields, you need to include it in the interface